src/go/rpk/pkg/cli/security/secret/create.go

```diff
@@ -20,15 +21,21 @@ import (
 	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/publicapi"
 	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
 )
 
+var vsm, scopeNames, scopeToText = createScopeMap()
```

Comment:
Let's try to split this function into pieces and use it where is needed instead outside of the Run function. Having it here will cause for rpk to run `createScopeMap` on every single command even if it's outside of the `secret packge`